core/tee/tee_svc_cryp.c

```diff
@@ -4253,14 +4420,18 @@ TEE_Result syscall_asymm_operate(unsigned long state,
 			    params[n].attributeID ==
 			    TEE_ATTR_RSA_OAEP_MGF_HASH) {
 				uint32_t hash = 0;
+				void *buf = params[n].content.ref.buffer;
 
 				if (params[n].content.ref.length !=
 				    sizeof(hash)) {
 					res = TEE_ERROR_BAD_PARAMETERS;
 					break;
 				}
-				memcpy(&hash, params[n].content.ref.buffer,
-				       sizeof(hash));
+
+				res = copy_from_user(&hash, buf, sizeof(hash));
+				if (res)
+					break;
```

Comment:
It seems a test on `res == SUCCESS` was missing on exit of the for loop (line 4442).
Or maybe replace the `break` instruction at lines 4428, 4433 and 4438 with a `goto out` instruction.

Reply:
updated the function to `goto out` if `copy_from_user()` fails, but left the other `break`s unchanged. Replacing the other two `break`s with `goto out` resulted in xtest failure. I guess those `break`s were intentionally used.